magicctapipe/io/gadf.py

```diff
@@ -0,0 +1,306 @@
+    theta_lo, theta_hi = split_bin_lo_hi(fov_offset_bins[np.newaxis, :].to(u.deg))
+
+    # Create a table
+    qtable = QTable(
+        data={
+            "ENERG_LO": energy_lo,
+            "ENERG_HI": energy_hi,
+            "THETA_LO": theta_lo,
+            "THETA_HI": theta_hi,
+            "GH_CUTS": gh_cuts.T[np.newaxis, :],
+        }
+    )
+
+    # Create a header
+    header = fits.Header(
+        cards=[
+            ("CREATOR", f"magicctapipe v{__version__}"),
+            ("HDUCLAS1", "RESPONSE"),
+            ("HDUCLAS2", "GH_CUTS"),
+            ("HDUCLAS3", "POINT-LIKE"),
+            ("HDUCLAS4", "GH_CUTS_2D"),
+            ("DATE", Time.now().utc.iso),
+        ]
+    )
+
+    for key, value in header_cards.items():
+        header[key] = value
+
+    # Create a HDU
+    gh_cuts_hdu = fits.BinTableHDU(qtable, header=header, name="GH_CUTS")
+
+    return gh_cuts_hdu
+
+
+@u.quantity_input(on_time=u.s, source_ra=u.deg, source_dec=u.deg)
+def create_event_hdu(
+    event_data, on_time, deadc, source_name, source_ra=None, source_dec=None
+):
+    """
+    Creates a fits binary table HDU for shower events.
+
+    Parameters
+    ----------
+    event_data: astropy.table.table.QTable
+        Table of the DL2 events surviving gammaness cuts
+    on_time: astropy.table.table.QTable
+        ON time of the input data
+    deadc: float
+        Dead time correction factor
+    source_name: str
+        Name of the observed source
+    source_ra: astropy.units.quantity.Quantity
+        Right ascension of the observed source
+        (Used only when the source name cannot be resolved)
+    source_dec: astropy.units.quantity.Quantity
+        Declination of the observed source
+        (Used only when the source name cannot be resolved)
+
+    Returns
+    -------
+    event_hdu: astropy.io.fits.hdu.table.BinTableHDU
+        Event HDU
+    """
+
+    mjdreff, mjdrefi = np.modf(MJDREF.mjd)
+
+    time_start = Time(event_data["timestamp"][0], format="unix", scale="utc")
+    time_start_iso = time_start.to_value("iso", "date_hms")
+
+    time_end = Time(event_data["timestamp"][-1], format="unix", scale="utc")
+    time_end_iso = time_end.to_value("iso", "date_hms")
+
+    elapsed_time = time_end - time_start
+    effective_time = on_time * deadc
+
+    event_coords = SkyCoord(
+        ra=event_data["reco_ra"], dec=event_data["reco_dec"], frame="icrs"
+    )
+
+    event_coords = event_coords.galactic
+
+    try:
+        # Try to get the coordinate from the source name
+        source_coord = SkyCoord.from_name(source_name, frame="icrs")
+
+    except Exception:
+        # Use the input RA/Dec coordinate instead
+        logger.warning(
+            f"WARNING: The source name '{source_name}' could not be resolved. "
+            f"Setting the input RA/Dec coordinate ({source_ra}, {source_dec})..."
+        )
+        source_coord = SkyCoord(ra=source_ra, dec=source_dec, frame="icrs")
```

Comment:
maybe add here an error message if from_name  failed and source_{ra,dec} is set to None?

Reply:
Thanks for your comment. I pushed a commit to add this error message now.